### ISSUE TITLE
test: --enable-gcc-warnings vs. time_t on 32bit platforms

### ISSUE THREAD

**Issue by jbeich** [libevent/libevent#448]
FreeBSD i386 and powerpc define `time_t` as `int32_t`:
```c
test/regress_util.c:1416:5: error: implicit conversion from 'long long' to 'time_t' (aka 'int') changes value from 4294967296 to 0 [-Werror,-Wconstant-conversion]
        {  4294967296, "Sun, 07 Feb 2106 06:28:16 GMT"} /* 2^32 */,
        ~  ^~~~~~~~~~
test/regress_util.c:1418:3: error: implicit conversion from 'long long' to 'time_t' (aka 'int') changes value from 253402300799 to -769665 [-Werror,-Wconstant-conversion]
        {253402300799, "Fri, 31 Dec 9999 23:59:59 GMT"} /* long long future no one can imagine */,
        ~^~~~~~~~~~~~
```
but FreeBSD armv6 defines `time_t` as `int64_t`:
```c
test/bench.c:203:4: error: format specifies type 'long' but the argument has type 'long long' [-Werror,-Wformat]
                        tv->tv_sec * 1000000L + tv->tv_usec);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test/bench_cascade.c:180:4: error: format specifies type 'long' but the argument has type 'long long' [-Werror,-Wformat]
                        tv->tv_sec * 1000000L + tv->tv_usec);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by jbeich** [comment]
FreeBSD mips (o32) defines `time_t` as `int64_t` (aka `long long`):
```
test/regress_util.c:1416: warning: integer constant is too large for 'long' type
test/regress_util.c:1418: warning: integer constant is too large for 'long' type
test/bench.c:203: warning: format '%ld' expects type 'long int', but argument 3 has type 'time_t'
test/bench_cascade.c:180: warning: format '%ld' expects type 'long int', but argument 3 has type 'time_t'
```

**Comment by azat** [comment]
First two issues (in regression tests) should be fixed by #554
